.github/workflows/nodejs.yml

```diff
@@ -6,6 +6,9 @@ on:
       - main
   pull_request:
 
+env:
+  npm_config_engine_strict: true
```

Comment:
[note] For `npm_config_` variables, ref https://docs.npmjs.com/cli/v11/using-npm/config#environment-variables